### PR TITLE
[Feature] New Privilege: support authorization for analyze statements

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Support authorization for analyze statements.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

- [ ] I have checked the version labels which the pr will be auto backported to target branch

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 108 (03.70%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|0|52|00.00%|[131, 132, 134, 135, 208, 209, 210, 211, 212, 213, 214, 215, 218, 219, 220, 221, 222, 223, 224, 225, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 238, 241, 527, 528, 529, 534, 535, 536, 537, 538, 539, 544, 545, 546, 551, 552, 553, 560, 567, 574, 581, 589]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|30|00.00%|[905, 906, 907, 908, 909, 911, 913, 916, 920, 921, 922, 923, 924, 925, 926, 927, 928, 930, 931, 932, 934, 935, 936, 937, 938, 939, 940, 941, 943, 947]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|2|00.00%|[65, 69]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|12|00.00%|[1679, 1680, 1681, 1698, 1699, 1700, 1716, 1717, 1718, 1734, 1735, 1736]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|3|33.33%|[65, 66]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|1|3|33.33%|[58, 59]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|1|3|33.33%|[73, 74]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|1|3|33.33%|[53, 54]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 81 / 107 (75.70%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|5|00.00%|[904, 905, 906, 907, 910]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|12|00.00%|[1679, 1680, 1681, 1698, 1699, 1700, 1716, 1717, 1718, 1734, 1735, 1736]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|67|76|88.16%|[134, 135, 137, 138, 601, 608, 615, 622, 630]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 1]
```diff
@@ -180,6 +207,81 @@ static void checkStmtNodePrivilege(ConnectContext context) {
...
+            List<Long> dbIds = GlobalStateMgr.getCurrentState().getDbIds();
+            for (Long id : dbIds) {
+                addTableNamesInDb(tableNames, id);
+            }
+        }
+
+        return tableNames;
+    }
+
+    private static void addTableNamesInDb(Set<TableName> tableNames, Long id) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:233]: How about name as `getTableNamesInDb`? `addTableNamesInDb` seems a ddl operation.

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:233]: good idea, actually this is a function generated by Intellij and i didn't look it carefully...

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Review by gengjun-git — DISMISSED** [review]

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 81 / 107 (75.70%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|5|00.00%|[904, 905, 906, 907, 910]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|12|00.00%|[1679, 1680, 1681, 1698, 1699, 1700, 1716, 1717, 1718, 1734, 1735, 1736]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|67|76|88.16%|[134, 135, 137, 138, 601, 608, 615, 622, 630]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|3|3|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 81 / 103 (78.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|5|00.00%|[904, 905, 906, 907, 910]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|12|00.00%|[1679, 1680, 1681, 1698, 1699, 1700, 1716, 1717, 1718, 1734, 1735, 1736]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|67|72|93.06%|[593, 600, 607, 614, 622]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|3|3|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by dulong41** [comment]
run starrocks_admit_test

**Pull request merged**